File: ArrayAndString/747. Largest Number At Least Twice of Others.java
class Solution {
    public int dominantIndex(int[] nums) {
        //fast but not the most space efficient
        int second = 0, maxVal = 0, maxIndex=0;
        for(int i=0; i<nums.length; i++){
            if(nums[i] > maxVal){
                second = maxVal;
                maxVal = nums[i];
                maxIndex = i;
            }
            else if(nums[i]>second) second = nums[i];
        }
        if(maxVal >= second*2) return maxsIndex;

        return -1;
    }
}

File: ArrayAndString/747. Largest Number At Least Twice of Others.py
class Solution:
    def dominantIndex(self, nums: List[int]) -> int:
        second = 0
        maxVal = 0
        for val in nums:
            if (val > maxVal):
                second = maxVal
                maxVal = val
            elif(val > second):
                second = val

        if(maxVal >= 2*second): return nums.index(maxVal)

        return -1
        """
        second = 0
        maxVal = 0
        maxIndex = 0
        for index,val in enumerate(nums):
            if (val > maxVal):
                second = maxVal
                maxVal = val
                maxIndex = index
            elif(val > second):
                second = val

        if(maxVal >= 2*second): return maxIndex

        return -1
        """